crates/interpreter/src/valid.rs

```diff
@@ -412,24 +406,43 @@ struct Expr<'a, 'm> {
     is_body: bool,
     locals: Vec<ValType>,
     labels: Vec<Label<'m>>,
+    // TODO(dev/fast-interp): Remove `Option` after `side_table` is verified.
+    side_table: Vec<Option<SideTableEntryView>>,
+}
+
+#[derive(Debug, Copy, Clone)]
+struct SideTableBranch<'m> {
+    parser: &'m [u8],
+    side_table: usize,
+}
+
+impl SideTableBranch<'_> {
+    fn delta_ip(&self, source: &[u8]) -> i32 {
+        (self.parser.as_ptr() as isize - source.as_ptr() as isize) as i32
+    }
+
+    fn delta_stp(&self, source: usize) -> i32 {
+        (self.side_table - source) as i32
+    }
```

Comment:
Actually, let's inline those 2 functions in `stitch` to incentivize using `stitch`. Ideally, all stitching should occur there.

Reply:
(I didn't do this refactor due to the borrow checker error mentioned in another comment.)